app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ContactsController < ApplicationController
+  include GoogleAuthentication
+  include GoogleEventClient
+
+  before_action :refresh_token
+  before_action :check_authentication
+
+  def index
+    render json: users.to_json
+  end
+
+  private
+
+  def users
+    GoogleContacts.new(session[:credentials]).call.users || []
```

Comment:
I think that you are not catching exceptions. I believe that it sometimes throw something :)